dm-manager/pkg/dm/amt.go

```diff
@@ -5,65 +5,333 @@ package dm
 
 import (
 	"context"
+	"fmt"
+	"strings"
 	"sync"
 	"time"
 
+	inventoryv1 "github.com/open-edge-platform/infra-core/inventory/v2/pkg/api/inventory/v1"
+	"github.com/open-edge-platform/infra-core/inventory/v2/pkg/client"
 	"github.com/open-edge-platform/infra-core/inventory/v2/pkg/logging"
+	"github.com/open-edge-platform/infra-core/inventory/v2/pkg/secretprovider"
 	"github.com/open-edge-platform/infra-external/dm-manager/pkg/api/mps"
+	"github.com/open-edge-platform/infra-external/dm-manager/pkg/api/rps"
+)
+
+const (
+	mpsPort          = 4433
+	fqdnServerFormat = 201
+	passwordAuth     = 2
+
+	AmtPasswordSecretName = "amt-password"
+	passwordKey           = "password"
+
+	StaticPasswordPolicy  = "static"
+	DynamicPasswordPolicy = "dynamic"
 )
 
 var log = logging.GetLogger("DmReconciler")
 
+type ReconcilerConfig struct {
+	ClusterDomain  string
+	PasswordPolicy string
+
+	RequestTimeout  time.Duration
+	ReconcilePeriod time.Duration
+}
 type Reconciler struct {
-	APIClient mps.ClientWithResponsesInterface
-	TermChan  chan bool
-	ReadyChan chan bool
-	WaitGroup *sync.WaitGroup
+	MpsClient       mps.ClientWithResponsesInterface
+	RpsClient       rps.ClientWithResponsesInterface
+	InventoryClient client.TenantAwareInventoryClient
+	EventsWatcher   chan *client.WatchEvents
+	SecretProvider  secretprovider.SecretProvider
+	TermChan        chan bool
+	ReadyChan       chan bool
+	WaitGroup       *sync.WaitGroup
+	Config          *ReconcilerConfig
 }
 
 func (dmr *Reconciler) Start() {
-	ticker := time.NewTicker(time.Minute)
+	ticker := time.NewTicker(dmr.Config.ReconcilePeriod)
 	if dmr.ReadyChan != nil {
 		dmr.ReadyChan <- true
 	}
 	log.Info().Msgf("Starting periodic reconciliation")
-	dmr.Reconcile(context.Background())
+	dmr.ReconcileAdd()
+	dmr.ReconcileRemove()
 	for {
 		select {
 		case <-ticker.C:
 			log.Info().Msgf("Running periodic reconciliation")
-			dmr.Reconcile(context.Background())
+			dmr.ReconcileAdd()
+			dmr.ReconcileRemove()
 		case <-dmr.TermChan:
 			log.Info().Msgf("Stopping periodic reconciliation")
 			ticker.Stop()
 			dmr.WaitGroup.Done()
 			return
+		case event := <-dmr.EventsWatcher:
+			if event.Event.GetEventKind() == inventoryv1.SubscribeEventsResponse_EVENT_KIND_CREATED {
+				log.Info().Msgf("Received create event: %v", event.Event.GetResource().GetTenant().GetResourceId())
+				dmr.handleTenantCreation(event.Event.GetResource().GetTenant().GetTenantId())
+			}
+
+			if event.Event.GetEventKind() == inventoryv1.SubscribeEventsResponse_EVENT_KIND_DELETED {
```

Comment:
if you want to reach when the DESIRED_STATE is deleted you have to listen for UPDATE events.